src/components/slider/daikin-slider.ts

```diff
@@ -0,0 +1,321 @@
+  private _internals = this.attachInternals();
+
+  private _updateFormValue() {
+    if (this.disabled) {
+      return;
+    }
+    this._internals.setFormValue(this.value);
+    this.dispatchEvent(
+      new Event("change", {
+        bubbles: true,
+        composed: true,
+      })
+    );
+    this.dispatchEvent(
+      new Event("input", {
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  private get _progress(): number {
+    const min = parseFloat(this.min);
+    const max = parseFloat(this.max);
+    const clampedValue = Math.max(min, Math.min(parseFloat(this.value), max));
+    return ((clampedValue - min) / (max - min)) * 100;
+  }
+
+  // This function will triggered when click the slider bar area.
+  private _handleClick(event: MouseEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+    const leftDistance = this.calcMousePositionRatio(event);
+    if (leftDistance == null) {
+      return;
+    }
+    const value = getValueFromRatio(this, leftDistance);
+    this.value = value;
+  }
+
+  // This function will triggered when the slider thumb button be focused and click keyboard arrow key.
+  private _handleKeyDown(event: KeyboardEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+    const value = getValueByKeydown(this, event.key);
+    if (value == null) {
+      return;
+    }
+    this.value = value;
+  }
+
+  /**
+   * Returns the normalized position of the mouse on the slider, within the range 0 to 1.
+   *
+   * @param event A MouseEvent.
+   * @returns The mouse's X coordinate normalized to a range of 0 to 1. The left end and beyond of the slider is 0, and the right end and beyond is 1.
+   */
+  private calcMousePositionRatio(
+    event: MouseEvent | TouchEvent
+  ): number | undefined {
+    const sliderRef = this._sliderRef.value?.getBoundingClientRect();
+    if (sliderRef == null) {
+      return;
+    }
+    const left =
+      event instanceof MouseEvent
+        ? event.clientX - sliderRef.left
+        : event.targetTouches[0].clientX - sliderRef.left;
+    const newLeft = Math.max(0, Math.min(left, sliderRef.width));
+    return newLeft / sliderRef.width;
+  }
+
+  // This function will triggered when the slider thumb button start dragging.
+  private _startDrag(event: MouseEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+
+    const onDrag = (event: MouseEvent) => {
+      const leftDistance = this.calcMousePositionRatio(event);
+      if (leftDistance == null) {
+        return;
+      }
+      const value = getValueFromRatio(this, leftDistance);
+      this.value = value;
+    };
+
+    const stopDrag = () => {
+      document.removeEventListener("mousemove", onDrag);
+      document.removeEventListener("mouseup", stopDrag);
+    };
+
+    document.addEventListener("mousemove", onDrag);
+    document.addEventListener("mouseup", stopDrag);
```

Comment:
1. Reuse drag handler.
2. Move slider on first mouse press.

```suggestion
    const stopDrag = (): void => {
      document.removeEventListener("mousemove", this._handleDrag);
      document.removeEventListener("mouseup", stopDrag);
    };

    document.addEventListener("mousemove", this._handleDrag);
    document.addEventListener("mouseup", stopDrag);

    this._handleDrag(event);
```